src/style.js

```diff
@@ -1,20 +1,22 @@
 import { Component } from 'react'
-import StyleSheetRegistry from './stylesheet-registry'
-
-const styleSheetRegistry = new StyleSheetRegistry()
+import { computeId, StyleSheetRegistryContext } from './stylesheet-registry'
 
 export default class JSXStyle extends Component {
   constructor(props) {
     super(props)
     this.prevProps = {}
   }
 
+  get styleSheetRegistry() {
+    return this.context
+  }
+
   static dynamic(info) {
     return info
       .map(tagInfo => {
         const baseId = tagInfo[0]
         const props = tagInfo[1]
-        return styleSheetRegistry.computeId(baseId, props)
+        return computeId(baseId, props)
```

Comment:
Is there any reason for not passing a `cache` here? Probably it is ok to export a version with a built in cache (the iife) I wrote in the other comment.

Reply:
I can do that. My only worry is just that it might become a bit of a memory leak since you have a cache entry per unique set of props (so if you have a long running app that uses different interpolations, they'd never get cleared).

What do you think?

Reply:
Oh well it has been like this for ever and nobody has reported any issue of this kind. If necessary we can replace the object with a WeakMap any time though (later)

Reply:
I'd say let's fix this (bring back the cache)